Last test case started to fail locally. here is the output:
```
• [FAILED] [122.262 seconds]
[sig-node] DRA [Feature:DynamicResourceAllocation][NodeFeature:DynamicResourceAllocation] Resource Kubelet Plugin [Serial] [It] must process pod created when kubelet is not running
test/e2e_node/dra_test.go:94

  Timeline >>
  STEP: Creating a kubernetes client @ 06/01/23 17:09:33.552
  STEP: Building a namespace api object, basename dra-node @ 06/01/23 17:09:33.553
  Jun  1 17:09:33.558: INFO: Skipping waiting for service account
  STEP: start Kubelet plugin @ 06/01/23 17:09:33.562
  I0601 17:09:33.562753   38402 nonblockinggrpcserver.go:105] "kubelet plugin/dra: GRPC server started" node="devel"
  I0601 17:09:33.562898   38402 nonblockinggrpcserver.go:105] "kubelet plugin/registrar: GRPC server started" node="devel"
  I0601 17:09:34.607296   38402 nonblockinggrpcserver.go:116] "kubelet plugin/registrar: handling request" node="devel" requestID=1 request="&InfoRequest{}"
  I0601 17:09:34.607351   38402 nonblockinggrpcserver.go:127] "kubelet plugin/registrar: handling request succeeded" node="devel" requestID=1 response="&PluginInfo{Type:DRAPlugin,Name:test-driver.cdi.k8s.io,Endpoint:/var/lib/kubelet/plugins/test-driver/dra.sock,SupportedVersions:[1.0.0],}"
  I0601 17:09:34.607901   38402 nonblockinggrpcserver.go:116] "kubelet plugin/registrar: handling request" node="devel" requestID=2 request="&RegistrationStatus{PluginRegistered:true,Error:,}"
  I0601 17:09:34.607944   38402 nonblockinggrpcserver.go:127] "kubelet plugin/registrar: handling request succeeded" node="devel" requestID=2 response="&RegistrationStatusResponse{}"
  Jun  1 17:09:35.587: INFO: Get running kubelet with systemctl:   UNIT                            LOAD   ACTIVE SUB     DESCRIPTION
    kubelet-20230601T170928.service loaded active running /home/ed/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet --kubeconfig /home/ed/go/src/k8s.io/kubernetes/_output/local/go/bin/kubeconfig --root-dir /var/lib/kubelet --v 4 --feature-gates DynamicResourceAllocation=true --hostname-override devel --container-runtime-endpoint unix:///run/containerd/containerd.sock --config /home/ed/go/src/k8s.io/kubernetes/_output/local/go/bin/kubelet-config --cluster-domain=cluster.local

  LOAD   = Reflects whether the unit definition was properly loaded.
  ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
  SUB    = The low-level unit activation state, values depend on unit type.
  1 loaded units listed.
  , kubelet-20230601T170928
  I0601 17:09:35.953536   38402 nonblockinggrpcserver.go:116] "kubelet plugin/registrar: handling request" node="devel" requestID=3 request="&InfoRequest{}"
  I0601 17:09:35.953583   38402 nonblockinggrpcserver.go:127] "kubelet plugin/registrar: handling request succeeded" node="devel" requestID=3 response="&PluginInfo{Type:DRAPlugin,Name:test-driver.cdi.k8s.io,Endpoint:/var/lib/kubelet/plugins/test-driver/dra.sock,SupportedVersions:[1.0.0],}"
  I0601 17:09:35.954292   38402 nonblockinggrpcserver.go:116] "kubelet plugin/registrar: handling request" node="devel" requestID=4 request="&RegistrationStatus{PluginRegistered:true,Error:,}"
  I0601 17:09:35.954328   38402 nonblockinggrpcserver.go:127] "kubelet plugin/registrar: handling request succeeded" node="devel" requestID=4 response="&RegistrationStatusResponse{}"
  I0601 17:09:36.907526   38402 nonblockinggrpcserver.go:116] "kubelet plugin/dra: handling request" node="devel" requestID=1 request="&NodePrepareResourceRequest{Namespace:dra-node-8314,ClaimUid:87ef47c3-091d-4bba-884b-340f2c171b1e,ClaimName:external-claim,ResourceHandle:{\"EnvVars\":{\"DRA_PARAM1\":\"PARAM1_VALUE\"},\"NodeName\":\"\"},}"
  I0601 17:09:36.907889   38402 kubeletplugin.go:195] "kubelet plugin/dra: CDI file created" node="devel" requestID=1 path="/var/run/cdi/test-driver.cdi.k8s.io-87ef47c3-091d-4bba-884b-340f2c171b1e.json" device="test-driver.cdi.k8s.io/test=claim-87ef47c3-091d-4bba-884b-340f2c171b1e"
  I0601 17:09:36.907936   38402 nonblockinggrpcserver.go:127] "kubelet plugin/dra: handling request succeeded" node="devel" requestID=1 response="&NodePrepareResourceResponse{CdiDevices:[test-driver.cdi.k8s.io/test=claim-87ef47c3-091d-4bba-884b-340f2c171b1e],}"
  I0601 17:09:39.312286   38402 nonblockinggrpcserver.go:116] "kubelet plugin/dra: handling request" node="devel" requestID=2 request="&NodeUnprepareResourceRequest{Namespace:dra-node-8314,ClaimUid:87ef47c3-091d-4bba-884b-340f2c171b1e,ClaimName:external-claim,ResourceHandle:{\"EnvVars\":{\"DRA_PARAM1\":\"PARAM1_VALUE\"},\"NodeName\":\"\"},}"
  I0601 17:09:39.312367   38402 kubeletplugin.go:209] "kubelet plugin/dra: CDI file removed" node="devel" requestID=2 path="/var/run/cdi/test-driver.cdi.k8s.io-87ef47c3-091d-4bba-884b-340f2c171b1e.json"
  I0601 17:09:39.312400   38402 nonblockinggrpcserver.go:127] "kubelet plugin/dra: handling request succeeded" node="devel" requestID=2 response="&NodeUnprepareResourceResponse{}"
  Jun  1 17:11:35.737: INFO: Failed inside E2E framework:
      k8s.io/kubernetes/test/e2e/framework/pod.WaitForPodCondition({0x7efc68271000, 0xc000d47560}, {0x5e34a80?, 0xc001501d40?}, {0xc000bb51c0, 0xd}, {0x55178b7, 0x6}, {0xc000eda7b0, 0x13}, ...)
      	test/e2e/framework/pod/wait.go:227 +0x25f
      k8s.io/kubernetes/test/e2e/framework/pod.WaitForPodSuccessInNamespaceTimeout({0x7efc68271000, 0xc000d47560}, {0x5e34a80, 0xc001501d40}, {0x55178b7, 0x6}, {0xc000bb51c0, 0xd}, 0xd?)
      	test/e2e/framework/pod/wait.go:402 +0x177
      k8s.io/kubernetes/test/e2e_node.glob..func14.1.4({0x7efc68271000, 0xc000d47560})
      	test/e2e_node/dra_test.go:106 +0x1c5
  [FAILED] in [It] - test/e2e_node/dra_test.go:107 @ 06/01/23 17:11:35.737
  Jun  1 17:11:35.737: INFO: Waiting up to 7m0s for all (but 0) nodes to be ready
  I0601 17:11:35.752774   38402 nonblockinggrpcserver.go:101] "kubelet plugin/registrar: GRPC server terminated gracefully" node="devel"
  I0601 17:11:35.752818   38402 nonblockinggrpcserver.go:148] "kubelet plugin/registrar: GRPC server stopped" node="devel"
  I0601 17:11:35.752924   38402 nonblockinggrpcserver.go:101] "kubelet plugin/dra: GRPC server terminated gracefully" node="devel"
  I0601 17:11:35.752952   38402 nonblockinggrpcserver.go:148] "kubelet plugin/dra: GRPC server stopped" node="devel"
  STEP: dump namespace information after failure @ 06/01/23 17:11:35.753
  STEP: Collecting events from namespace "dra-node-8314". @ 06/01/23 17:11:35.753
  STEP: Found 4 events. @ 06/01/23 17:11:35.756
  Jun  1 17:11:35.756: INFO: At 2023-06-01 17:09:36 +0300 EEST - event for drapod: {kubelet devel} MissingClusterDNS: pod: "drapod_dra-node-8314(fee701a5-2e22-4a31-abae-ebb0feab02fb)". kubelet does not have ClusterDNS IP configured and cannot create Pod using "ClusterFirst" policy. Falling back to "Default" policy.
  Jun  1 17:11:35.756: INFO: At 2023-06-01 17:09:37 +0300 EEST - event for drapod: {kubelet devel} Pulled: Container image "registry.k8s.io/e2e-test-images/busybox:1.29-4" already present on machine
  Jun  1 17:11:35.756: INFO: At 2023-06-01 17:09:37 +0300 EEST - event for drapod: {kubelet devel} Created: Created container testcontainer
  Jun  1 17:11:35.756: INFO: At 2023-06-01 17:09:37 +0300 EEST - event for drapod: {kubelet devel} Started: Started container testcontainer
  Jun  1 17:11:35.758: INFO: POD  NODE  PHASE  GRACE  CONDITIONS
  Jun  1 17:11:35.758: INFO: 
  Jun  1 17:11:35.761: INFO: 
  Logging node info for node devel
  Jun  1 17:11:35.764: INFO: Node Info: &Node{ObjectMeta:{devel    26e9bdb9-c2e9-4229-940a-24aa9a2d25dd 101 0 2023-06-01 17:09:28 +0300 EEST <nil> <nil> map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:devel kubernetes.io/os:linux] map[volumes.kubernetes.io/controller-managed-attach-detach:true] [] [] [{kubelet Update v1 2023-06-01 17:09:28 +0300 EEST FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:volumes.kubernetes.io/controller-managed-attach-detach":{}},"f:labels":{".":{},"f:beta.kubernetes.io/arch":{},"f:beta.kubernetes.io/os":{},"f:kubernetes.io/arch":{},"f:kubernetes.io/hostname":{},"f:kubernetes.io/os":{}}}} } {kubelet Update v1 2023-06-01 17:09:36 +0300 EEST FieldsV1 {"f:status":{"f:conditions":{"k:{\"type\":\"DiskPressure\"}":{"f:lastHeartbeatTime":{}},"k:{\"type\":\"MemoryPressure\"}":{"f:lastHeartbeatTime":{}},"k:{\"type\":\"PIDPressure\"}":{"f:lastHeartbeatTime":{}},"k:{\"type\":\"Ready\"}":{"f:lastHeartbeatTime":{},"f:message":{},"f:reason":{},"f:status":{}}}}} status}]},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{cpu: {{24 0} {<nil>} 24 DecimalSI},ephemeral-storage: {{1006450962432 0} {<nil>} 982862268Ki BinarySI},hugepages-1Gi: {{2147483648 0} {<nil>} 2Gi BinarySI},hugepages-2Mi: {{536870912 0} {<nil>}  BinarySI},memory: {{67119329280 0} {<nil>} 65546220Ki BinarySI},pods: {{110 0} {<nil>} 110 DecimalSI},},Allocatable:ResourceList{cpu: {{24 0} {<nil>} 24 DecimalSI},ephemeral-storage: {{905805864690 0} {<nil>} 905805864690 DecimalSI},hugepages-1Gi: {{2147483648 0} {<nil>} 2Gi BinarySI},hugepages-2Mi: {{536870912 0} {<nil>}  BinarySI},memory: {{64172830720 0} {<nil>} 62668780Ki BinarySI},pods: {{110 0} {<nil>} 110 DecimalSI},},Phase:,Conditions:[]NodeCondition{NodeCondition{Type:MemoryPressure,Status:False,LastHeartbeatTime:2023-06-01 17:09:35 +0300 EEST,LastTransitionTime:2023-06-01 17:09:28 +0300 EEST,Reason:KubeletHasSufficientMemory,Message:kubelet has sufficient memory available,},NodeCondition{Type:DiskPressure,Status:False,LastHeartbeatTime:2023-06-01 17:09:35 +0300 EEST,LastTransitionTime:2023-06-01 17:09:28 +0300 EEST,Reason:KubeletHasNoDiskPressure,Message:kubelet has no disk pressure,},NodeCondition{Type:PIDPressure,Status:False,LastHeartbeatTime:2023-06-01 17:09:35 +0300 EEST,LastTransitionTime:2023-06-01 17:09:28 +0300 EEST,Reason:KubeletHasSufficientPID,Message:kubelet has sufficient PID available,},NodeCondition{Type:Ready,Status:True,LastHeartbeatTime:2023-06-01 17:09:35 +0300 EEST,LastTransitionTime:2023-06-01 17:09:28 +0300 EEST,Reason:KubeletReady,Message:kubelet is posting ready status. AppArmor enabled,},},Addresses:[]NodeAddress{NodeAddress{Type:InternalIP,Address:10.237.72.187,},NodeAddress{Type:Hostname,Address:devel,},},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:10250,},},NodeInfo:NodeSystemInfo{MachineID:f53561508e7d4eb6917278f1b5e6f30f,SystemUUID:03d502e0-045e-05e3-a906-200700080009,BootID:e5613bcd-70ec-44cd-8330-49a39044d71b,KernelVersion:5.15.0-69-generic,OSImage:Ubuntu 22.04.2 LTS,ContainerRuntimeVersion:containerd://1.6.21,KubeletVersion:v1.28.0-alpha.0.1306+f8944d18638941-dirty,KubeProxyVersion:v1.28.0-alpha.0.1306+f8944d18638941-dirty,OperatingSystem:linux,Architecture:amd64,},Images:[]ContainerImage{ContainerImage{Names:[docker.io/library/perl@sha256:8977843c6dccf7105add6234d728627f1120bf5f181119c528ee855fa2925c86 docker.io/library/perl:5.34],SizeBytes:337594724,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/perl@sha256:c613344cdd31c5055961b078f831ef9d9199fc9111efe6e81bea3f00d78bd979 registry.k8s.io/e2e-test-images/perl:5.26],SizeBytes:325397679,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/node-perf/tf-wide-deep@sha256:91ab3b5ee22441c99370944e2e2cb32670db62db433611b4e3780bdee6a8e5a1 registry.k8s.io/e2e-test-images/node-perf/tf-wide-deep:1.3],SizeBytes:213443508,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/volume/gluster@sha256:033a12fe65438751690b519cebd4135a3485771086bcf437212b7b886bb7956c registry.k8s.io/e2e-test-images/volume/gluster:1.3],SizeBytes:115217395,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/jessie-dnsutils@sha256:24aaf2626d6b27864c29de2097e8bbb840b3a414271bf7c8995e431e47d8408e registry.k8s.io/e2e-test-images/jessie-dnsutils:1.7],SizeBytes:112030336,},ContainerImage{Names:[registry.k8s.io/ingress-nginx/controller@sha256:34ee929b111ffc7aa426ffd409af44da48e5a0eea1eb2207994d9e0c0882d143],SizeBytes:107049438,},ContainerImage{Names:[registry.k8s.io/etcd@sha256:e013d0d5e4e25d00c61a7ff839927a1f36479678f11e49502b53a5e0b14f10c3 registry.k8s.io/etcd:3.5.9-0],SizeBytes:102894559,},ContainerImage{Names:[registry.k8s.io/etcd@sha256:b3a0da984f9af598234c018faa4f28dd6a6461f5719163a61f10ebf26dbca1e9 registry.k8s.io/etcd:3.5.8-0],SizeBytes:102889341,},ContainerImage{Names:[registry.k8s.io/etcd@sha256:dd75ec974b0a2a6f6bb47001ba09207976e625db898d1b16735528c009cb171c registry.k8s.io/etcd:3.5.6-0],SizeBytes:102542580,},ContainerImage{Names:[registry.k8s.io/etcd@sha256:6f72b851544986cb0921b53ea655ec04c36131248f16d4ad110cb3ca0c369dc1 registry.k8s.io/etcd:3.5.4-0],SizeBytes:102157811,},ContainerImage{Names:[registry.k8s.io/etcd@sha256:51eae8381dcb1078289fa7b4f3df2630cdc18d09fb56f8e56b41c40e191d6c83 registry.k8s.io/etcd:3.5.7-0],SizeBytes:101639218,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/volume/nfs@sha256:3bda73f2428522b0e342af80a0b9679e8594c2126f2b3cca39ed787589741b9e registry.k8s.io/e2e-test-images/volume/nfs:1.3],SizeBytes:95836203,},ContainerImage{Names:[registry.k8s.io/sig-storage/nfs-provisioner@sha256:e943bb77c7df05ebdc8c7888b2db289b13bf9f012d6a3a5a74f14d4d5743d439 registry.k8s.io/sig-storage/nfs-provisioner:v3.0.1],SizeBytes:90632047,},ContainerImage{Names:[quay.io/kubevirt/device-plugin-kvm@sha256:b44bc0fd6ff8987091bbc7ec630e5ee6683be40d151b4e6635e24afb5807b21a quay.io/kubevirt/device-plugin-kvm:latest],SizeBytes:87743847,},ContainerImage{Names:[docker.io/kubernetesui/dashboard@sha256:290bebc3cd96c22b6f89e7b21f5c2b16ce5c275a0ec2c2de10e0d8b9dd110289 docker.io/kubernetesui/dashboard:v2.6.1],SizeBytes:75789999,},ContainerImage{Names:[docker.io/library/nginx@sha256:6202beb06ea61f44179e02ca965e8e13b961d12640101fca213efbfd145d7575 docker.io/library/nginx:1.10],SizeBytes:71402657,},ContainerImage{Names:[docker.io/library/nginx:1.9],SizeBytes:71180964,},ContainerImage{Names:[docker.io/library/httpd@sha256:5fa96551b61359de5dfb7fd8c9e97e4153232eb520a8e883e2f47fc80dbfc33e],SizeBytes:57131601,},ContainerImage{Names:[docker.io/library/httpd@sha256:87a012bf99bf5e3e0f628ac1f69abbeab534282857fba3a359ca3a3f4a02429a docker.io/library/httpd:latest],SizeBytes:57087633,},ContainerImage{Names:[docker.io/library/nginx@sha256:b8f2383a95879e1ae064940d9a200f67a6c79e710ed82ac42263397367e7cc4e docker.io/library/nginx:latest],SizeBytes:56882371,},ContainerImage{Names:[docker.io/library/nginx@sha256:943c25b4b66b332184d5ba6bb18234273551593016c0e0ae906bab111548239f],SizeBytes:56841090,},ContainerImage{Names:[docker.io/library/nginx:1.8],SizeBytes:54862786,},ContainerImage{Names:[docker.io/library/nginx@sha256:df13abe416e37eb3db4722840dd479b00ba193ac6606e7902331dcea50f4f1f2 docker.io/library/nginx:1.19],SizeBytes:53736948,},ContainerImage{Names:[docker.io/library/nginx@sha256:6b5f5eec0ac03442f3b186d552ce895dce2a54be6cb834358040404a242fd476 docker.io/library/nginx:1.19.9],SizeBytes:53729997,},ContainerImage{Names:[docker.io/library/nginx@sha256:b0ea179ab61c789ce759dbe491cc534e293428ad232d00df83ce44bf86261179 docker.io/library/nginx:1.19.8],SizeBytes:53693052,},ContainerImage{Names:[docker.io/library/nginx@sha256:e90ac5331fe095cea01b121a3627174b2e33e06e83720e9a934c7b8ccc9c55a0 docker.io/library/nginx:1.18 docker.io/library/nginx:1.18.0],SizeBytes:53630764,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/agnhost@sha256:2c5b5b056076334e4cf431d964d102e44cbca8f1e6b16ac1e477a0ffbe6caac4 registry.k8s.io/e2e-test-images/agnhost:2.45],SizeBytes:52363605,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/agnhost@sha256:16bbf38c463a4223d8cfe4da12bc61010b082a79b4bb003e2d3ba3ece5dd5f9e registry.k8s.io/e2e-test-images/agnhost:2.43],SizeBytes:51706353,},ContainerImage{Names:[registry.k8s.io/node-problem-detector/node-problem-detector@sha256:0ce71ef6d759425d22b10e65b439749fe5d13377a188e2fc060b731cdb4e6901 registry.k8s.io/node-problem-detector/node-problem-detector:v0.8.7],SizeBytes:47786458,},ContainerImage{Names:[docker.io/library/nginx@sha256:23b4dcdf0d34d4a129755fc6f52e1c6e23bb34ea011b315d87e193033bcd1b68 docker.io/library/nginx:1.15],SizeBytes:44795472,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/httpd@sha256:148b022f5c5da426fc2f3c14b5c0867e58ef05961510c84749ac1fddcb0fef22 registry.k8s.io/e2e-test-images/httpd:2.4.38-4],SizeBytes:40764257,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/node-perf/npb-is@sha256:8285539c79625b192a5e33fc3d21edc1a7776fb9afe15fae3b5037a7a8020839 registry.k8s.io/e2e-test-images/node-perf/npb-is:1.2],SizeBytes:39706325,},ContainerImage{Names:[registry.k8s.io/e2e-test-images/node-perf/npb-ep@sha256:90b5cfc5451428aad4dd6af9960640f2506804d35aa05e83c11bf0a46ac318c8 registry.k8s.io/e2e-test-images/node-perf/npb-ep:1.2],SizeBytes:39705373,},ContainerImage{Names:[registry.k8s.io/kube-apiserver@sha256:b8dda58b0c680898b6ab7fdbd035a75065d3607a70c3c4986bc1d8cfba5f0ec8 registry.k8s.io/kube-apiserver:v1.26.3],SizeBytes:35358932,},ContainerImage{Names:[registry.k8s.io/kube-apiserver@sha256:0f03b93af45f39704b7da175db31e20da63d2ab369f350e59de8cbbef9d703e0 registry.k8s.io/kube-apiserver:v1.26.2],SizeBytes:35329425,},ContainerImage{Names:[registry.k8s.io/kube-apiserver@sha256:99e1ed9fbc8a8d36a70f148f25130c02e0e366875249906be0bcb2c2d9df0c26 registry.k8s.io/kube-apiserver:v1.26.1],SizeBytes:35320235,},ContainerImage{Names:[registry.k8s.io/kube-apiserver@sha256:4188262a351f156e8027ff81693d771c35b34b668cbd61e59c4a4490dd5c08f3 registry.k8s.io/kube-apiserver:v1.25.3],SizeBytes:34238163,},ContainerImage{Names:[gcr.io/gcp-runtimes/ubuntu_18_0_4@sha256:dd77a035b4bf904e9569efb68054a852eb8772ee2f5b0b91fdf79c2b620536bc gcr.io/gcp-runtimes/ubuntu_18_0_4:latest],SizeBytes:34166638,},ContainerImage{Names:[registry.k8s.io/kube-controller-manager@sha256:40adecbe3a40aa147c7d6e9a1f5fbd99b3f6d42d5222483ed3a47337d4f9a10b registry.k8s.io/kube-controller-manager:v1.26.1],SizeBytes:32245960,},ContainerImage{Names:[registry.k8s.io/kube-controller-manager@sha256:28c0deb96fd861af5560b7e24b989a5d61f4a4b8467852d6975bf2a3a5217840 registry.k8s.io/kube-controller-manager:v1.26.3],SizeBytes:32206603,},ContainerImage{Names:[registry.k8s.io/kube-controller-manager@sha256:5434d52f88eb16bc5e98ccb65e97e97cb5cf7861749afbf26174d27c4ece1fad registry.k8s.io/kube-controller-manager:v1.26.2],SizeBytes:32180749,},ContainerImage{Names:[registry.k8s.io/kube-controller-manager@sha256:d3a06262256f3e7578d5f77df137a8cdf58f9f498f35b5b56d116e8a7e31dc91 registry.k8s.io/kube-controller-manager:v1.25.3],SizeBytes:31261869,},ContainerImage{Names:[gcr.io/cadvisor/cadvisor@sha256:89e6137f068ded2e9a3a012ce71260b9afc57a19305842aa1074239841a539a7 gcr.io/cadvisor/cadvisor:v0.43.0],SizeBytes:31056802,},ContainerImage{Names:[docker.io/library/ubuntu@sha256:b6b83d3c331794420340093eb706a6f152d9c1fa51b262d9bf34594887c2c7ac docker.io/library/ubuntu:22.04],SizeBytes:30427123,},ContainerImage{Names:[registry.k8s.io/kube-proxy@sha256:d89b6c6a8ecc920753df713b268b0d226f795135c4a0ecc5ce61660e623dd6da registry.k8s.io/kube-proxy:v1.26.3],SizeBytes:21543879,},ContainerImage{Names:[registry.k8s.io/kube-proxy@sha256:5dac6611aceb1452a5d4036108a15ceb0699c083a942977e30640d521e7d2078 registry.k8s.io/kube-proxy:v1.26.2],SizeBytes:21541935,},ContainerImage{Names:[registry.k8s.io/kube-proxy@sha256:85f705e7d98158a67432c53885b0d470c673b0fad3693440b45d07efebcda1c3 registry.k8s.io/kube-proxy:v1.26.1],SizeBytes:21536169,},ContainerImage{Names:[registry.k8s.io/kube-proxy@sha256:6bf25f038543e1f433cb7f2bdda445ed348c7b9279935ebc2ae4f432308ed82f registry.k8s.io/kube-proxy:v1.25.3],SizeBytes:20265805,},ContainerImage{Names:[docker.io/kubernetesui/metrics-scraper@sha256:76049887f07a0476dc93efc2d3569b9529bf982b22d29f356092ce206e98765c docker.io/kubernetesui/metrics-scraper:v1.0.8],SizeBytes:19746404,},ContainerImage{Names:[registry.k8s.io/ingress-nginx/kube-webhook-certgen@sha256:39c5b2e3310dc4264d638ad28d9d1d96c4cbb2b2dcfb52368fe4e3c63f61e10f],SizeBytes:19670592,},},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:nil,},}
  Jun  1 17:11:35.765: INFO: 
  Logging kubelet events for node devel
  Jun  1 17:11:35.767: INFO: 
  Logging pods the kubelet thinks is on node devel
  W0601 17:11:35.781592   38402 metrics_grabber.go:111] Can't find any pods in namespace kube-system to grab metrics from
  Jun  1 17:11:35.811: INFO: 
  Latency metrics for node devel
  STEP: Destroying namespace "dra-node-8314" for this suite. @ 06/01/23 17:11:35.811
  << Timeline

  [FAILED] Timed out after 120.001s.
  The matcher passed to Eventually returned the following error:
      <*errors.errorString | 0xc00076a7c0>: 
      pod "drapod" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:10.237.72.187 PodIP:10.85.0.3 PodIPs:[{IP:10.85.0.3} {IP:1100:200::3}] StartTime:2023-06-01 17:09:36 +0300 EEST InitContainerStatuses:[] ContainerStatuses:[{Name:testcontainer State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-06-01 17:09:37 +0300 EEST,FinishedAt:2023-06-01 17:09:37 +0300 EEST,ContainerID:containerd://64a7ecbc860e392876de86bf9e3facbcf03d4d95f333795eb2b7e1c7deb64469,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://64a7ecbc860e392876de86bf9e3facbcf03d4d95f333795eb2b7e1c7deb64469 Started:0xc001a2a78e AllocatedResources:map[] Resources:nil}] QOSClass:BestEffort EphemeralContainerStatuses:[] Resize:}
      {
          s: "pod \"drapod\" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-06-01 17:09:36 +0300 EEST Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:10.237.72.187 PodIP:10.85.0.3 PodIPs:[{IP:10.85.0.3} {IP:1100:200::3}] StartTime:2023-06-01 17:09:36 +0300 EEST InitContainerStatuses:[] ContainerStatuses:[{Name:testcontainer State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-06-01 17:09:37 +0300 EEST,FinishedAt:2023-06-01 17:09:37 +0300 EEST,ContainerID:containerd://64a7ecbc860e392876de86bf9e3facbcf03d4d95f333795eb2b7e1c7deb64469,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://64a7ecbc860e392876de86bf9e3facbcf03d4d95f333795eb2b7e1c7deb64469 Started:0xc001a2a78e AllocatedResources:map[] Resources:nil}] QOSClass:BestEffort EphemeralContainerStatuses:[] Resize:}",
      }
  In [It] at: test/e2e_node/dra_test.go:107 @ 06/01/23 17:11:35.737
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:
  [FAIL] [sig-node] DRA [Feature:DynamicResourceAllocation][NodeFeature:DynamicResourceAllocation] Resource Kubelet Plugin [Serial] [It] must process pod created when kubelet is not running
  test/e2e_node/dra_test.go:107

Ran 3 of 424 Specs in 146.060 seconds
FAIL! -- 2 Passed | 1 Failed | 0 Pending | 421 Skipped
--- FAIL: TestE2eNode (146.08s)
FAIL

Ginkgo ran 1 suite in 2m26.190802943s

```

--- a/test/e2e_node/dra_test.go
+++ b/test/e2e_node/dra_test.go
@@ -1,0 +1,230 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+E2E Node test for DRA (Dynamic Resource Allocation)
+This test covers node-specific aspects of DRA
+The test can be run locally on Linux this way:
+  make test-e2e-node FOCUS='\[NodeFeature:DynamicResourceAllocation\]' SKIP='\[Flaky\]' PARALLELISM=1 \
+       TEST_ARGS='--feature-gates="DynamicResourceAllocation=true" --service-feature-gates="DynamicResourceAllocation=true" --runtime-config=api/all=true'
+*/
+
+package e2enode
+
+import (
+	"context"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	resourcev1alpha2 "k8s.io/api/resource/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	"k8s.io/dynamic-resource-allocation/kubeletplugin"
+	testdriver "k8s.io/kubernetes/test/e2e/dra/test-driver/app"
+)
+
+const (
+	driverName                = "test-driver.cdi.k8s.io"
+	cdiDir                    = "/var/run/cdi"
+	endpoint                  = "/var/lib/kubelet/plugins/test-driver/dra.sock"
+	pluginRegistrationPath    = "/var/lib/kubelet/plugins_registry"
+	draAddress                = "/var/lib/kubelet/plugins/test-driver/dra.sock"
+	pluginRegistrationTimeout = time.Second * 60 // how long to wait for a node plugin to be registered
+)
+
+var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation][NodeFeature:DynamicResourceAllocation]", func() {
+	f := framework.NewDefaultFramework("dra-node")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	var kubeletPlugin *testdriver.ExamplePlugin
+
+	ginkgo.Context("Resource Kubelet Plugin [Serial]", func() {
+		ginkgo.BeforeEach(func(ctx context.Context) {
+			kubeletPlugin = newKubeletPlugin(getNodeName(ctx, f))
+		})
+
+		ginkgo.It("must register after Kubelet restart", func(ctx context.Context) {
+			oldCalls := kubeletPlugin.GetGRPCCalls()
+			getNewCalls := func() []testdriver.GRPCCall {
+				calls := kubeletPlugin.GetGRPCCalls()
+				return calls[len(oldCalls):]
+			}
+
+			ginkgo.By("restarting Kubelet")
+			restartKubelet(true)
+
+			ginkgo.By("wait for Kubelet plugin re-registration")
+			gomega.Eventually(getNewCalls).WithTimeout(pluginRegistrationTimeout).Should(testdriver.BeRegistered)
+		})
+
+		ginkgo.It("must register after plugin restart", func(ctx context.Context) {
+			ginkgo.By("restart Kubelet Plugin")
+			kubeletPlugin.Stop()
+			kubeletPlugin = newKubeletPlugin(getNodeName(ctx, f))
+
+			ginkgo.By("wait for Kubelet plugin re-registration")
+			gomega.Eventually(kubeletPlugin.GetGRPCCalls).WithTimeout(pluginRegistrationTimeout).Should(testdriver.BeRegistered)
+		})
+
+		ginkgo.It("must process pod created when kubelet is not running", func(ctx context.Context) {
+			// Stop Kubelet
+			startKubelet := stopKubelet()
+			pod := createTestObjects(ctx, f.ClientSet, getNodeName(ctx, f), f.Namespace.Name, "draclass", "external-claim", "drapod")
+			// Pod must be in pending state
+			err := e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod.Name, "Pending", framework.PodStartShortTimeout, func(pod *v1.Pod) (bool, error) {
+				return pod.Status.Phase == v1.PodPending, nil
+			})
+			framework.ExpectNoError(err)
+			// Start Kubelet
+			startKubelet()
+			// Pod should succeed
+			err = e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, f.ClientSet, pod.Name, f.Namespace.Name, framework.PodStartShortTimeout)
+			framework.ExpectNoError(err)
+		})
+	})
+})
+
+// Run Kubelet plugin and wait until it's registered
+func newKubeletPlugin(nodeName string) *testdriver.ExamplePlugin {
+	ginkgo.By("start Kubelet plugin")
+	logger := klog.LoggerWithValues(klog.LoggerWithName(klog.Background(), "kubelet plugin"), "node", nodeName)
+
+	// Ensure that directories exist, creating them if necessary. We want
+	// to know early if there is a setup problem that would prevent
+	// creating those directories.
+	err := os.MkdirAll(cdiDir, os.FileMode(0750))
+	framework.ExpectNoError(err, "create CDI directory")
+	err = os.MkdirAll(filepath.Dir(endpoint), 0750)
+	framework.ExpectNoError(err, "create socket directory")
+
+	plugin, err := testdriver.StartPlugin(
+		logger,
+		cdiDir,
+		driverName,
+		"",
+		testdriver.FileOperations{},
+		kubeletplugin.PluginSocketPath(endpoint),
+		kubeletplugin.RegistrarSocketPath(path.Join(pluginRegistrationPath, driverName+"-reg.sock")),
+		kubeletplugin.KubeletPluginSocketPath(draAddress),
+	)
+	framework.ExpectNoError(err)
+
+	gomega.Eventually(plugin.GetGRPCCalls).WithTimeout(pluginRegistrationTimeout).Should(testdriver.BeRegistered)
+
+	ginkgo.DeferCleanup(plugin.Stop)
+
+	return plugin
+}
+
+// createTestObjects creates objects required by the test
+// NOTE: as scheduler and controller manager are not running by the Node e2e,
+// the objects must contain all required data to be processed correctly by the API server
+// and placed on the node without involving the scheduler and the DRA controller
+func createTestObjects(ctx context.Context, clientSet kubernetes.Interface, nodename, namespace, className, claimName, podName string) *v1.Pod {
+	// ResourceClass
+	class := &resourcev1alpha2.ResourceClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: className,
+		},
+		DriverName: driverName,
+	}
+	_, err := clientSet.ResourceV1alpha2().ResourceClasses().Create(ctx, class, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.DeferCleanup(clientSet.ResourceV1alpha2().ResourceClasses().Delete, className, metav1.DeleteOptions{})
+
+	// ResourceClaim
+	podClaimName := "resource-claim"
+	claim := &resourcev1alpha2.ResourceClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: claimName,
+		},
+		Spec: resourcev1alpha2.ResourceClaimSpec{
+			ResourceClassName: className,
+		},
+	}
+	createdClaim, err := clientSet.ResourceV1alpha2().ResourceClaims(namespace).Create(ctx, claim, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.DeferCleanup(clientSet.ResourceV1alpha2().ResourceClaims(namespace).Delete, claimName, metav1.DeleteOptions{})
+
+	// Pod
+	containerName := "testcontainer"
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			NodeName: nodename, // Assign the node as the scheduler is not running
+			ResourceClaims: []v1.PodResourceClaim{
+				{
+					Name: podClaimName,
+					Source: v1.ClaimSource{
+						ResourceClaimName: &claimName,
+					},
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name:  containerName,
+					Image: e2epod.GetDefaultTestImage(),
+					Resources: v1.ResourceRequirements{
+						Claims: []v1.ResourceClaim{{Name: podClaimName}},
+					},
+					Command: []string{"/bin/sh", "-c", "env | grep DRA_PARAM1=PARAM1_VALUE"},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	createdPod, err := clientSet.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.DeferCleanup(clientSet.CoreV1().Pods(namespace).Delete, podName, metav1.DeleteOptions{})
+
+	// Update claim status: set ReservedFor and AllocationResult
+	// NOTE: This is usually done by the DRA controller
+	createdClaim.Status = resourcev1alpha2.ResourceClaimStatus{
+		DriverName: driverName,
+		ReservedFor: []resourcev1alpha2.ResourceClaimConsumerReference{
+			{Resource: "pods", Name: podName, UID: createdPod.UID},
+		},
+		Allocation: &resourcev1alpha2.AllocationResult{
+			ResourceHandles: []resourcev1alpha2.ResourceHandle{
+				{
+					DriverName: driverName,
+					Data:       "{\"EnvVars\":{\"DRA_PARAM1\":\"PARAM1_VALUE\"},\"NodeName\":\"\"}",
+				},
+			},
+		},
+	}
+	_, err = clientSet.ResourceV1alpha2().ResourceClaims(namespace).UpdateStatus(ctx, createdClaim, metav1.UpdateOptions{})
+	framework.ExpectNoError(err)
+
+	return pod
+}

